src/app/georeferencer/qgsvectorwarper.h

```diff
@@ -0,0 +1,68 @@
+/***************************************************************************
+     qgsvectorwarper.h
+     --------------------------------------
+    Date                 :
+    Copyright            :
+    Email                :
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSVECTORWARPER_H
+#define QGSVECTORWARPER_H
+
+#include <QCoreApplication>
+#include <QString>
+#include "qgsgcpgeometrytransformer.h"
+
+#include <vector>
+
+class QgsCoordinateReferenceSystem;
+class QgsGCPList;
+class QWidget;
+class QgsFeedback;
+
+/**
+ * Base implementation of QgsVectorWarper.
+ * \param method QgsGcpTransofmer method, based on gdal transformation methods
+ * \param points data points used in the georeferencer
+ * \param destCrs destination projection of the vector layer
+ * \since QGIS 3.20
+ */
+class QgsVectorWarper
+{
+
+  public:
+    explicit QgsVectorWarper( QgsGcpTransformerInterface::TransformMethod &method, const QgsGCPList &points,
+                              const QgsCoordinateReferenceSystem &destCrs );
+
+    /**
+     * Functions to reproject features of the vector layer
+     * \param layer source vector layer
+     * \param feedback optional qgsfeedback instance
+     * \return True if operation finished properly, otherwise false.
+     * \since QGIS 3.20
```

Comment:
can you update all since and rebase the PR please ? Thanks

Reply:
Thanks Loic for de-staling my PR and taking your time to look at them. There is one pending issue in this PR that I recall and that's when creating the output, it always make a geopackage. I'd have to validate why the extension was always ignored and added a gpkg no matter what...